### PR TITLE
D8 1490 - ACCESS Organization ingest and API endpoint

### PR DESCRIPTION
**Pull request by zooley**
## Describe context / purpose for this PR

Add Drush command for ingesting ACCESS organizations (to an ACCESS Organization content type) and associated API endpoint for autocompletes, etc.

## Issue link

https://cyberteamportal.atlassian.net/browse/D8-1490

## Checklist for PR author
- [X] I have checked that the PR is ready to be merged
- [X] I have reviewed the DIFF and checked that the changes are as expected
- [X] I have assigned myself or someone else to review the PR

### CODE REVIEW

File: src/Commands/AccessCommands.php [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        $path = \Drupal::service('file_system')->realpath("private://") . '/.keys/secrets.json';
+        if (!file_exists($path)) {
+            if ($options['verbose']) {
+                $this->output()->writeln('<error>Unable to find ACCESS API key file</error>');
+            }
+            return;
+        }
+        $secrets_json_text = file_get_contents($path);
+        $secrets_data = json_decode($secrets_json_text, true);
+        if (!isset($secrets_data['access_api_key']) || !$secrets_data['access_api_key']) {
```
Comment (a-pasquale) [inline src/Commands/AccessCommands.php]: The api key is currently stored on the live filesystem with the 'ramps_api_key' key.

**Comment by a-pasquale** [comment]
I'm not seeing content items appearing in Drupal Content list. I thought this was resolved with the domain access issue? Maybe that got reverted somehow?

Also this issue with undefined variables:

Record already exists for "Exa Corporation".
 [warning] Undefined variable $lat AccessCommands.php:150
 [warning] Undefined variable $lon AccessCommands.php:155

**Comment by zooley** [comment]
Undefined variables: Oops. I was confused for a moment as I wasn't getting those warnings. I simply missed that file in the commit. It should be updated now.

As for the domain fix: The changes I made to the content type on production were gone. I re-applied them. Let's see if it sticks this time...

**Comment by a-pasquale** [comment]
> As for the domain fix: The changes I made to the content type on production were gone. I re-applied them. Let's see if it sticks this time...

Config changes should be added to the[ pull request](https://github.com/necyberteam/cyberteam_drupal/pull/657 ).

**Comment by zooley** [comment]
> Config changes should be added to thehttps://github.com/necyberteam/cyberteam_drupal/pull/657.

PR #657 is for the Campus Champions theme. This pertains to the ACCESS Organizations content type and is not related. The config for the added domain fields were actually committed by you (I remember this now as I had the commit staged and was going to push but noticed you had beat me to it by like an hour):

https://github.com/necyberteam/cyberteam_drupal/pull/635/commits/b996ac57703f4984a3159794a8420619678d934c

(Which was a followup from https://github.com/necyberteam/cyberteam_drupal/pull/620)

But that still doesn't explain why those fields disappeared from production. Is some form of re-sync being performed on prod?

**Comment by a-pasquale** [comment]
Oh, my mistake. I wasn't feeling well yesterday. I meant this pull request: https://github.com/necyberteam/cyberteam_drupal/pull/635

It hasn't been merge to main yet. I was waiting on the last things on this PR. 

The live/prod config gets overwritten whenever a new release is made, so any changes made on live also need to get added to the the repository.

**Pull request merged**